core/runtime/src/main/java/io/quarkiverse/langchain4j/RegisterAiService.java

```diff
@@ -88,11 +89,21 @@
     Class<? extends Supplier<ChatMemoryProvider>> chatMemoryProviderSupplier() default BeanChatMemoryProviderSupplier.class;
 
     /**
-     * Configures the way to obtain the {@link Retriever} to use (when using RAG). All tools are expected to be CDI beans
-     * By default, no supplier is used.
+     * Configures the way to obtain the {@link Retriever} to use (when using RAG).
+     * By default, no retriever is used.
+     *
+     * @deprecated Use retrievalAugmentor instead
      */
+    @Deprecated(forRemoval = true)
     Class<? extends Retriever<TextSegment>> retriever() default NoRetriever.class;
 
+    /**
+     * Configures the way to obtain the {@link RetrievalAugmentor} to use (when using RAG).
+     * The Supplier is expected to be a CDI bean.
+     * By default, no retrieval augmentor is used.
+     */
+    Class<? extends Supplier<RetrievalAugmentor>> retrievalAugmentor() default NoRetrievalAugmentor.class;
```

Comment:
For `Retriever`, the typical usage was creating a custom implementation of the `Retriever` interface, which is nicely doable by having the user provide a CDI bean that implements `Retriever`. 

But for `RetrievalAugmentor`, the typical usage is to use the builder of `DefaultRetrievalAugmentor` to build an appropriately configured instance of it. We don't want the user to create a class that extends `RetrievalAugmentor`, we want them to use the builder to build an instance of `DefaultRetrievalAugmentor`.

Reply:
Then in that case I would much prefer to use the approach we have with `moderationModelSupplier` (and others).

Reply:
That is pretty much what I did here, no? Or what exactly should be different?

Reply:
`moderationModelSupplier` is more flexible because:

It provides `BeanModerationModelSupplier` which essentially looks up the bean, but it also allows for an arbitraty supplier that does not need to be a bean (for advanced cases)

Reply:
Hmm, I'm still not completely convinced. You mean that 
- if the supplier class a CDI bean, then inject it from CDI
- if it's not a CDI bean, then call the no-arg constructor?

In 99% of cases, you need the supplier to be a bean, because you can't build a reasonable augmentor instance without injecting an embedding store and model. That way we would open up a doorway to user errors where they have injection points in their `Supplier<RetrievalAugmentor>`, but we call the constructor instead of using it as a CDI bean.

Reply:
The is the following:

* `BeanRetrievalAugmentorSupplier` would automatically look use whatever `BeanRetrievalAugmentor` CDI then is.
* Any other class would be a customer supplier would allow users to return whatever `BeanRetrievalAugmentor` they like from `get`. This class may or may not be a CDI bean.

Reply:
Maybe we're misunderstanding each other, I see no reason why the supplier implementation should be allowed to NOT be a bean :)

Reply:
Anyway I've still pushed an update where it is allowed to not be a bean, so please check it

Reply:
Thanks, I'll have a look tomorrow